samples/notifications/dialog/form/src/index.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import { IgrButton, IgrDialog, IgrInput, IgrIcon, IgrForm, IgrButtonModule, IgrDialogModule, IgrInputModule, IgrIconModule, IgrFormModule } from 'igniteui-react';
+import 'igniteui-webcomponents/themes/light/bootstrap.css';
+
+IgrButtonModule.register();
+IgrDialogModule.register();
+IgrInputModule.register();
+IgrIconModule.register();
+IgrFormModule.register();
+
+export default class DialogForm extends React.Component<any, any> {
+
+    public dialogRef: IgrDialog;
+    public passwordIcon: IgrIcon;
+    public usernameIcon: IgrIcon;
+
+    constructor(props: any) {
+        super(props);
+        this.onDialogRef = this.onDialogRef.bind(this);
+        this.onDialogShow = this.onDialogShow.bind(this);
+        this.iconPasswordRef = this.iconPasswordRef.bind(this); 
+        this.iconUsernameRef = this.iconUsernameRef.bind(this); 
+    }
+
+    public render(): JSX.Element {
+        return (
+            <div className="container sample center">
+                <IgrButton variant="contained" clicked={this.onDialogShow}>
+                    <span>Show Dialog</span>
+                </IgrButton>
+
+                <IgrDialog title="Login" ref={this.onDialogRef}>
+                    <IgrForm method="dialog">
+                        <IgrInput label="Username">
+                            <span slot="prefix">
+                                <IgrIcon ref={this.iconUsernameRef} name="username" collection="material"/>
+                            </span>
+                        </IgrInput>
+                        <br />
+                        <IgrInput type="password" label="Password">
+                            <span slot="prefix">
+                                <IgrIcon ref={this.iconPasswordRef} name="password" collection="material"/>
+                            </span>
+                        </IgrInput>
+                        <br />
+                        <div style={{display: "flex", justifyContent: "flex-end"}}>
+                            <IgrButton type="reset" variant="flat"><span>Reset</span></IgrButton>
+                            <IgrButton type="submit" variant="flat"><span>Submit</span></IgrButton>
```

Comment:
I've noticed an issue related to the form's reset and submit button functionalities in React. Currently, interacting with the reset button doesn't trigger the expected behavior, probably as a result of an event handling discrepancy. I would suspect that the same case would be with the submit button as well. This would best be addressed in a separate issue.

As for the present sample, we might consider implementing a temporary workaround that handles the reset button in the form.

Reply:
Work item for the reset/submit buttons: https://infragistics.visualstudio.com/NetAdvantage/_workitems/edit/28887